DependencyInjection/Configuration.php

```diff
@@ -49,103 +50,88 @@ public function addScopesConfiguration(ArrayNodeDefinition $rootNode): self
     {
         $rootNode
             ->fixXmlConfig('scope')
-                ->children()
-                    ->arrayNode('scopes')
-                        ->scalarPrototype()
-                    ->end()
-                ->end()
-            ->end();
+            ->children()
+            ->arrayNode('scopes')
+            ->scalarPrototype();
 
         return $this;
     }
 
     public function addKeyChainsConfiguration(ArrayNodeDefinition $rootNode): self
     {
-        $rootNode
+        $keyChainsIdentifier = $rootNode
             ->fixXmlConfig('key_chain')
-                ->children()
-                    ->arrayNode('key_chains')
-                        ->useAttributeAsKey('identifier')
-                        ->arrayPrototype()
-                        ->children()
-                            ->scalarNode('key_set_name')->isRequired()->end()
-                            ->scalarNode('public_key')->isRequired()->end()
-                            ->scalarNode('private_key')->defaultNull()->end()
-                            ->scalarNode('private_key_passphrase')->defaultNull()->end()
-                            ->scalarNode('algorithm')->defaultValue(KeyInterface::ALG_RS256)->end()
-                        ->end()
-                    ->end()
-                ->end()
-            ->end();
+            ->children()
+            ->arrayNode('key_chains')
+            ->useAttributeAsKey('identifier')
+            ->arrayPrototype()
+            ->children();
+
+        $keyChainsIdentifier->scalarNode('key_set_name')->isRequired();
```

Comment:
if it's for psalm, I would put in balance the clear loss of readability vs psalm compliance

Reply:
the psalm is the problem in this case, I will include the file in the ignore file, this is not the best, but now it is necessary.

Reply:
yeah if you dont mind imo it's more useful to keep the config tree understandable